ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/PostCommitValidation.scala

```diff
@@ -170,18 +164,49 @@ private[appendonlydao] object PostCommitValidation {
 
   private type Result = Either[RejectionReason, State]
 
+  /** The active ledger key state during validation.
+    * After a rollback node, we restore the state at the
+    * beginning of the rollback.
+    *
+    * @param contracts Active contracts created in
+    *  the current transaction that have a key indexed
+    *  by a hash of their key.
+    * @param removed Hashes of contract keys that are known to
+    *  to be archived. Note that a later create with the same
+    *  key will remove the entry again.
+    */
+  private final case class ActiveState(
+      val contracts: Map[Hash, ContractId],
+      val removed: Set[Hash],
```

Comment:
```suggestion
      contracts: Map[Hash, ContractId],
      removed: Set[Hash],
```
`val` is implicit for all fields of a `case class`.